app/models/manageiq/providers/autosde/inventory/collector/target_collection.rb

```diff
@@ -8,49 +8,51 @@ def initialize(_manager, _target)
   def physical_storages
     return [] if references(:physical_storages).blank?
 
-    super
+    @physical_storages ||= @manager.autosde_client.StorageSystemApi.storage_systems_get.select { |s| references(:physical_storages).include?(s.uuid) }
   end
 
   def storage_resources
     return [] if references(:storage_resources).blank?
 
-    super
+    @storage_resources ||= @manager.autosde_client.StorageResourceApi.storage_resources_get.select { |s| references(:storage_resources).include?(s.uuid) }
   end
 
   def storage_hosts
     return [] if references(:storage_hosts).blank?
-
-    super
   end
```

Comment:
For these that aren't implemented yet best to just return `[]` since currently this will return `nil` if there are any storage_host targets